examples/distributed-training/trl/README.md

```diff
@@ -0,0 +1,184 @@
+        --num_machines=$DSTACK_NODES_NUM 
+        trl/scripts/sft.py
+        --model_name $MODEL_ID
+        --dataset_name OpenAssistant/oasst_top1_2023-08-25
+        --dataset_text_field="text"
+        --per_device_train_batch_size 1
+        --per_device_eval_batch_size 1
+        --gradient_accumulation_steps 4 
+        --learning_rate 2e-4
+        --report_to wandb
+        --bf16
+        --max_seq_length 1024
+        --attn_implementation flash_attention_2 
+        --logging_steps=10
+        --output_dir /checkpoints/llama31-ft
+        --hub_model_id $HUB_MODEL_ID
+        --torch_dtype bfloat16
+
+    resources:
+      gpu: 80GB:8
+      shm_size: 128GB
+
+    volumes:
+      - /checkpoints:/checkpoints
+    ```
+    </div>
+
+=== "Deepseed ZeRO-3"
+
+    <div editor-title="examples/distributed-training/trl/deepspeed.dstack.yml">
+    ```yaml
+    type: task
+    # The name is optional, if not specified, generated randomly
+    name: trl-train-deepspeed-distrib
+
+    # Size of the cluster
+    nodes: 2
+
+    image: nvcr.io/nvidia/pytorch:25.01-py3
+
+    # Required environment variables
+    env:
+      - HF_TOKEN
+      - ACCELERATE_LOG_LEVEL=info
+      - WANDB_API_KEY
+      - MODEL_ID=meta-llama/Llama-3.1-8B
+      - HUB_MODEL_ID
+    
+    # Commands of the task
+    commands:
+      - pip install transformers
+      - pip install bitsandbytes
+      - pip install peft
+      - pip install wandb
+      - pip install deepspeed
+      - git clone https://github.com/huggingface/trl
+      - cd trl
+      - pip install .
+      - accelerate launch
+        --config_file=examples/accelerate_configs/deepspeed_zero3.yaml
+        --main_process_ip=$DSTACK_MASTER_NODE_IP
+        --main_process_port=8008
+        --machine_rank=$DSTACK_NODE_RANK 
+        --num_processes=$DSTACK_GPUS_NUM
+        --num_machines=$DSTACK_NODES_NUM
+        trl/scripts/sft.py
+        --model_name $MODEL_ID
+        --dataset_name OpenAssistant/oasst_top1_2023-08-25
+        --dataset_text_field="text"
+        --per_device_train_batch_size 1
+        --per_device_eval_batch_size 1
+        --gradient_accumulation_steps 4 
+        --learning_rate 2e-4
+        --report_to wandb
+        --bf16
+        --max_seq_length 1024
+        --attn_implementation flash_attention_2
+        --logging_steps=10
+        --output_dir /checkpoints/llama31-ft
+        --hub_model_id $HUB_MODEL_ID
+        --torch_dtype bfloat16
+
+    resources:
+      gpu: 80GB:8
+      shm_size: 128GB
+
+    volumes:
+      - /checkpoints:/checkpoints
+    ```
+    </div>
+
+
+!!! Note
+    We are using the NGC container because it includes the necessary libraries and packages for RDMA and InfiniBand support.
+
+### Applying the configuration
+To run a configuration, use the [`dstack apply`](https://dstack.ai/docs/reference/cli/dstack/apply.md) command.
+
+<div class="termy">
+
+```shell
+$ dstack apply -f examples/distributed-training/trl/fsdp.dstack.yml
```

Comment:
Need to set environment variables which values aren't configured in YAML (`HF_TOKEN`, `WANDB_API_KEY`, etc).